packages/@lwc/engine-core/src/framework/def.ts

```diff
@@ -137,12 +137,26 @@ function createComponentDef(Ctor: LightningElementConstructor): ComponentDef {
         decoratorsMeta;
     const proto = Ctor.prototype;
 
+    const publicProperties = keys(apiFields);
+    const protoDescriptors = Object.getOwnPropertyDescriptors(proto);
```

Comment:
`getOwnPropertyDescriptors` can be pulled in from `@lwc/shared`. (We typically try to pull these kinds of methods from `@lwc/shared` for historical reasons – namely bundle size and preventing someone from polluting globals after LWC is loaded – although I admit we are not 100% consistent about it.)